src/frontend/components/UI/AuthModal/index.tsx

```diff
@@ -0,0 +1,160 @@
+import React, { useEffect, useRef } from 'react'
+import styles from './index.module.scss'
+import { ModalAnimation } from '@hyperplay/ui'
+import { WebviewTag } from 'electron'
+import { observer } from 'mobx-react-lite'
+import authState from '../../../state/authState'
+import { ethers } from 'ethers'
+import extensionState from '../../../state/ExtensionState'
+import onboardingState from '../../../store/OnboardingStore'
+import walletState from '../../../state/WalletState'
+import { DEV_PORTAL_URL } from 'common/constants'
+
+const url = `${DEV_PORTAL_URL}/signin`
+
+const METAMASK_ALREADY_PROVIDED_ERROR_CODE = -32002
```

Comment:
For later, we can always have a file where that includes Metamask error codes at